lib/model/query/entities.js

```diff
@@ -23,13 +23,16 @@ const { isTrue } = require('../../util/http');
 // Submission-defining entitiies contain the dataset name as a string
 // so we will look up the dataset id by name (unique within a project)
 // and project id (from submission def -> submission -> form def -> form)
-const _getDataset = (dsName, subDefId) => sql`
-SELECT datasets."id", sd."id" as "subDefId", datasets."acteeId", datasets."name"
-FROM submission_defs AS sd
+const _getDataset = (dsName, subDefId, dsId) => sql`
```

Comment:
Explanation for reviewer: this subquery is important for the case where the entity is parsed from the submission and we don't know the dataset until we look it up by name and project (found from subdef -> sub -> form -> project). Obviously in this other case, we have the dataset more directly and we don't have a submission def.